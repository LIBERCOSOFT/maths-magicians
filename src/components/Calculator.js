import React from 'react';
import calculate from '../logic/calculate';

export class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
  }

  handleClick = (e) => {
    this.setState((state) => calculate(state, e.target.innerText));
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <>
        <div className="display">
          {' '}
          {total}
          {' '}
          {operation}
          {' '}
          {next}
          {' '}
        </div>

        <div className="buttonContainer">
          <div>
            <button type="button" onClick={this.handleClick}>AC</button>
            <button type="button" onClick={this.handleClick}>+/-</button>
            <button type="button" onClick={this.handleClick}>%</button>
            <button type="button" onClick={this.handleClick}>+</button>
          </div>
          <div>
            <button type="button" onClick={this.handleClick}>7</button>
            <button type="button" onClick={this.handleClick}>8</button>
            <button type="button" onClick={this.handleClick}>9</button>
            <button type="button" onClick={this.handleClick}>x</button>
          </div>
          <div>
            <button type="button" onClick={this.handleClick}>4</button>
            <button type="button" onClick={this.handleClick}>5</button>
            <button type="button" onClick={this.handleClick}>6</button>
            <button type="button" onClick={this.handleClick}>-</button>
          </div>
          <div>
            <button type="button" onClick={this.handleClick}>1</button>
            <button type="button" onClick={this.handleClick}>2</button>
            <button type="button" onClick={this.handleClick}>3</button>
            <button type="button" onClick={this.handleClick}>+</button>
          </div>
          <div>
            <button type="button" className="zero" onClick={this.handleClick}>0</button>
            <button type="button" onClick={this.handleClick}>.</button>
            <button type="button" onClick={this.handleClick}>=</button>
          </div>
        </div>

      </>
    );
  }
}

export default Calculator;
