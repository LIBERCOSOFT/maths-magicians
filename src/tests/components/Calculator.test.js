import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Calculator from '../../pages/Calculator';

test('renders calculator component snapshot', () => {
  const { asFragment } = render(<Calculator />, { wrapper: BrowserRouter });
  expect(asFragment()).toMatchSnapshot();
});
