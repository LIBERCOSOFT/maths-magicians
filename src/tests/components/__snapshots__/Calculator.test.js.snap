// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders calculator component snapshot 1`] = `
<DocumentFragment>
  <section
    class="calculator__container"
  >
    <header>
      <h2>
        Let's do some math!
      </h2>
    </header>
    <div>
      <div
        class="display"
      >
         0   
      </div>
      <div
        class="buttonContainer"
      >
        <div>
          <button
            type="button"
          >
            AC
          </button>
          <button
            type="button"
          >
            +/-
          </button>
          <button
            type="button"
          >
            %
          </button>
          <button
            type="button"
          >
            รท
          </button>
        </div>
        <div>
          <button
            type="button"
          >
            7
          </button>
          <button
            type="button"
          >
            8
          </button>
          <button
            type="button"
          >
            9
          </button>
          <button
            type="button"
          >
            x
          </button>
        </div>
        <div>
          <button
            type="button"
          >
            4
          </button>
          <button
            type="button"
          >
            5
          </button>
          <button
            type="button"
          >
            6
          </button>
          <button
            type="button"
          >
            -
          </button>
        </div>
        <div>
          <button
            type="button"
          >
            1
          </button>
          <button
            type="button"
          >
            2
          </button>
          <button
            type="button"
          >
            3
          </button>
          <button
            type="button"
          >
            +
          </button>
        </div>
        <div>
          <button
            class="zero"
            type="button"
          >
            0
          </button>
          <button
            type="button"
          >
            .
          </button>
          <button
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
